include/isotp.h

```diff
@@ -0,0 +1,348 @@
+#endif
+
+/**
+ * @brief ISO-TP message id struct
+ *
+ * Used to pass addresses to the bind and send functions.
+ */
+struct iso_tp_msg_id {
+	/** Message identifier*/
+	union {
+		u32_t std_id  : 11;
+		u32_t ext_id  : 29;
+	};
+	/** extended address */
+	u8_t ext_addr;
+	/** Indicates the identifier type (standard or extended) */
+	u8_t id_type : 1;
+	/** Indicates if extended addressing is used */
+	u8_t use_ext_addr : 1;
+};
+
+/*
+ * STmin is split in two valid ranges:
+ *   0-127: 0ms-127ms
+ * 128-240: Reserved
+ * 241-249: 100us-900us (multiples of 100us)
+ * 250-   : Reserved
+ */
+struct iso_tp_fc_opts {
+	u8_t bs;     /** Block size. Number of CF PDUs before next CF is sent */
+	u8_t stmin;  /** Minimum separation time. Min time between frames */
+};
+
+typedef void (*isotp_tx_callback_t)(int error_nr, void *arg);
+
+struct iso_tp_send_ctx;
+struct iso_tp_recv_ctx;
+
+/**
+ * @brief Bind an address to a receiving context.
+ *
+ * This function binds a RX and TX address combination to a RX context.
+ * When data arrives on the specified address it is buffered and can the
+ * be read by calling iso_tp_rcv.
+ * When calling this routine, a filter is applied in the CAN device and the
+ * context is initialized. The contest must be valid until calling unbind.
+ *
+ * @param ctx     Context to store the internal states.
+ * @param can_dev The CAN device to be used for sending and receiving.
+ * @param rx_addr Identifier for incoming data.
+ * @param tx_addr Identifier for FC frames.
+ * @param opts    Flow control options.
+ * @param timeout Timeout for FF SF buffer allocation.
+ *
+ * @retval ISO_TP_N_OK on success,
+ *         ISO_TP_NO_FREE_FILTER if CAN device has no filters left.
+ */
+int iso_tp_bind(struct iso_tp_recv_ctx *ctx, struct device *can_dev,
+		const struct iso_tp_msg_id *rx_addr,
+		const struct iso_tp_msg_id *tx_addr,
+		const struct iso_tp_fc_opts *opts,
+		s32_t timeout);
+
+/**
+ * @brief Unbind a context from the interface
+ *
+ * This function removes the binding from iso_tp_bind.
+ * When calling this routine the filter is detached from the CAN device and
+ * if a transmission is ongoing, buffers are freed.
+ * The context can be discarded safely after this.
+ *
+ * @param ctx     Context that should be unbound.
+ */
+void iso_tp_unbind(struct iso_tp_recv_ctx *ctx);
+
+/**
+ * @brief Read out received data from fifo.
+ *
+ * This function blocking reads the data from the receive fifo of the context.
+ * If an error occurs the function returns a negative number and leaves the
+ * data buffer unchanged.
+ *
+ * @param ctx     Context that is already bound.
+ * @param data    Pointer to a buffer where the data is copied to.
+ * @param len     Size of the buffer.
+ * @param timeout Timeout for incoming data.
+ *
+ * @retval ISO_TP_N_OK on success, ISO_TP_WAIT_TIMEOUT when "timeout" timed out
+ * or any of ISO_TP_N_* on error
+ */
+int iso_tp_rcv(struct iso_tp_recv_ctx *ctx, u8_t *data, size_t len, s32_t timeout);
```

Comment:
Could we change this to `iso_tp_recv()`?

Reply:
I was proposing to call the receive function `iso_tp_recv()` instead `iso_tp_rcv()`, my reasoning being that recv is commonly used as a name when receiving things. Same way you already have `iso_tp_send()` instead `iso_tp_snd()`. Using recv is just more readable IMHO.

Reply:
Done.